sympy/ntheory/tests/test_residue.py

```diff
@@ -284,3 +286,33 @@ def test_residue():
     raises(ValueError, lambda: polynomial_congruence(x**x, 6125))
     raises(ValueError, lambda: polynomial_congruence(x**i, 6125))
     raises(ValueError, lambda: polynomial_congruence(0.1*x**2 + 6, 100))
+
+    assert binomial_mod(-1, 1, 10) == 0
+    assert binomial_mod(1, -1, 10) == 0
+    raises(ValueError, lambda: binomial_mod(2, 1, -1))
+    assert binomial_mod(51, 10, 10) == 0
+    assert binomial_mod(10**3, 500, 3**6) == 567
+    assert binomial_mod(10**18 - 1, 123456789, 4) == 0
+    assert binomial_mod(10**18, 10**12, (10**5 + 3)**2) == 3744312326
+
+
+@slow
```

Comment:
Is this slow because of `n` or just because some inputs take a long time to compute? If it's only due to `n`, we should just lower `n` so that we can run this always. The tests are randomized so a failure would be caught eventually (persisting failures is another question, but that's another thing that would be much easier to do with hypothesis).

Reply:
Actually I meant `trials` here, not `n`.

Reply:
It's mainly `trials` I believe, but slightly `n` as well, as a bigger `n` means bigger primes are chosen on average.

I've set the trials back to 1000 (and `n`) and removed `@slow`, it's still around a second, but this is still quicker than the other test in the file.

Reply:
I don't know if we have specific guidance on what should be marked as @slow, but a second sounds fine to leave it off as far as I'm concerned.